internal/services/appservice/windows_web_app_slot_resource.go

```diff
@@ -0,0 +1,616 @@
+	StorageAccounts               []helpers.StorageAccount              `tfschema:"storage_account"`
+	ConnectionStrings             []helpers.ConnectionString            `tfschema:"connection_string"`
+	CustomDomainVerificationId    string                                `tfschema:"custom_domain_verification_id"`
+	DefaultHostname               string                                `tfschema:"default_hostname"`
+	Kind                          string                                `tfschema:"kind"`
+	OutboundIPAddresses           string                                `tfschema:"outbound_ip_addresses"`
+	OutboundIPAddressList         []string                              `tfschema:"outbound_ip_address_list"`
+	PossibleOutboundIPAddresses   string                                `tfschema:"possible_outbound_ip_addresses"`
+	PossibleOutboundIPAddressList []string                              `tfschema:"possible_outbound_ip_address_list"`
+	SiteCredentials               []helpers.SiteCredential              `tfschema:"site_credential"`
+	Tags                          map[string]string                     `tfschema:"tags"`
+}
+
+var _ sdk.ResourceWithUpdate = WindowsWebAppSlotResource{}
+
+func (r WindowsWebAppSlotResource) ModelObject() interface{} {
+	return &WindowsWebAppSlotModel{}
+}
+
+func (r WindowsWebAppSlotResource) ResourceType() string {
+	return "azurerm_windows_web_app_slot"
+}
+
+func (r WindowsWebAppSlotResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.WebAppSlotID
+}
+
+func (r WindowsWebAppSlotResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.WebAppName,
+		},
+
+		"app_service_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.WebAppName,
+		},
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"location": location.Schema(),
+
+		"service_plan_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.ServicePlanID,
+		},
+
+		// Optional
+
+		"app_settings": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"auth_settings": helpers.AuthSettingsSchema(),
+
+		"backup": helpers.BackupSchema(),
+
+		"client_affinity_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"client_certificate_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"client_certificate_mode": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Default:  string(web.ClientCertModeRequired),
+			ValidateFunc: validation.StringInSlice([]string{
+				string(web.ClientCertModeOptional),
+				string(web.ClientCertModeRequired),
+			}, false),
+		},
```

Comment:
this default value doesn't look correct? iirc this is unset by default?

Reply:
IIRC this is the default value returned by the service, however, this setting does nothing until `client_certificate_enabled` is set to `true`. I will double check and update as appropriate.

Reply:
Yep, service default is `Required`

![image](https://user-images.githubusercontent.com/11830746/148513567-74d320ad-169c-42ae-bf9d-43193b052408.png)